networking_cisco/plugins/cisco/cfg_agent/service_helpers/routing_svc_helper_aci.py

```diff
@@ -185,5 +195,7 @@ def _internal_network_removed(self, ri, port, ex_gw_port):
         driver.internal_network_removed(ri, port,
                                         itfc_deleted=itfc_deleted)
         if ri.snat_enabled and ex_gw_port:
+            # We need the inverted logic for NAT
+            is_deleted = not itfc_deleted
```

Comment:
I had the original logic wrong when I added this parameter. I'll look into fixing the original logic instead, as I agree this is just confusing.